### PR TITLE
Decoupling multithreaded sts

### LINKED ISSUE #107: SymbolTableStack - Decoupling from Multithreaded Implementation
### Details
The current form of `SymbolTableStack` bakes in multithreading into its capabiliites using `Arc<Mutex<>>`. Remove this.

### Issue Type
- [X] Feature
- [ ] Bugfix
- [ ] Documentation
- [X] Refactoring
- [ ] Other:

### Requirements
- Decouple multithreading from `SymbolTableStack`
- Unit testing

### Notes

### PR DESCRIPTION
**Pull request by Dalyj24**
This closes #107

### CODE REVIEW

File: frontend/src/symbol_table/symbol_table_struct.rs
Comment (CalebLItalien) [inline frontend/src/symbol_table/symbol_table_struct.rs:1]: I think you have some dead code in  `get_element()` with `visualization`? You can keep this code in a debug function.

**Review by CalebLItalien — COMMENTED** [review]
Looks good! One comment below.

**Review by CalebLItalien — APPROVED** [review]
Approved assuming debug works.

**Comment by Dalyj24** [comment]
Closed issue #121 STS visualization was done with a function other than fmt::Display, which was fixed.

**Pull request merged**